src/lib.rs

```diff
@@ -323,6 +396,668 @@ const EAX_CACHE_PARAMETERS_AMD: u32 = 0x8000_001D;
+            self.source.set_leaf(
+                EAX_VENDOR_INFO,
+                Some(CpuIdResult {
+                    eax: 0,
+                    ebx: leaf.ebx,
+                    ecx: leaf.ecx,
+                    edx: leaf.edx,
+                }),
+            );
+        } else {
+            self.source.set_leaf(EAX_VENDOR_INFO, None);
+        }
+        Ok(())
+    }
+
+    /// Set extended function info (LEAF=0x8000_0000).
+    ///
+    /// Similar to leaf 0h, this leaf describes limit of valid extended CPUID leaves (at or above
+    /// 0x8000_0000) in EAX, with a vendor identifier (typically the same as leaf 0h) in EBX, ECX,
+    /// EDX.
+    ///
+    /// While the name is different, the leaf is structurally identical to leaf 0h (vendor info),
+    /// so that structure is taken to define this leaf as well.
+    ///
+    /// # Platforms
+    /// ✅ AMD ✅ Intel
+    pub fn set_extended_function_info(&mut self, leaf: Option<VendorInfo>) -> Result<(), ()> {
+        if let Some(leaf) = leaf {
+            self.source.set_leaf(
+                EAX_EXTENDED_FUNCTION_INFO,
+                Some(CpuIdResult {
+                    eax: 0,
+                    ebx: leaf.ebx,
+                    ecx: leaf.ecx,
+                    edx: leaf.edx,
+                }),
+            );
+        } else {
+            self.source.set_leaf(EAX_EXTENDED_FUNCTION_INFO, None);
+        }
+
+        Ok(())
+    }
+
+    /// Query a set of features that are available on this CPU (LEAF=0x01).
+    ///
+    /// # Platforms
+    /// ✅ AMD ✅ Intel
+    pub fn set_feature_info(&mut self, leaf: Option<FeatureInfo>) -> Result<(), ()> {
+        if let Some(leaf) = leaf {
+            let edx_ecx = leaf.edx_ecx.bits();
+            let bytes = edx_ecx.to_le_bytes();
+            use core::convert::TryInto;
+            let ecx = u32::from_le_bytes(bytes[0..4].try_into().expect("four bytes"));
+            let edx = u32::from_le_bytes(bytes[4..8].try_into().expect("four bytes"));
+            self.source.set_leaf(
+                EAX_FEATURE_INFO,
+                Some(CpuIdResult {
+                    eax: leaf.eax,
+                    ebx: leaf.ebx,
+                    ecx,
+                    edx,
+                }),
+            );
+        } else {
+            self.source.set_leaf(EAX_FEATURE_INFO, None);
+        }
+        Ok(())
+    }
+
+    /*
+     * TODO:
+     * set_cache_info (leaf 2h)
+     * set_processor_serial (leaf 3h)
+     * set_cache_parameters (leaf 4h)
+     * set_extended_topology_info_v2 (leaf 1Fh)
+     * set_rdt_monitoring_info (leaf Fh)
+     * set_rdt_allocation_info (leaf 10H)
+     * set_sgx_info (leaf 12h)
+     * set_processor_trace_info (leaf 14h)
+     * set_tsc_info (leaf 15h)
+     * set_processor_frequency_info (leaf 16h)
+     * set_soc_vendor_info (leaf 17h)
+     * set_deterministic_address_translation_info (leaf 18h)
+     * set_hypervisor_info
+     */
```

Comment:
my very AMD-specific use case meant I punted on all the Intel-only leaves but adding some setters here is straightforward if the change seems interesting.